src/contexts/field-focus.js

```diff
@@ -1,39 +1,107 @@
-import React, { createContext, useContext, useEffect, useState } from 'react';
+import React, { createContext, useContext, useEffect, useReducer } from 'react';
 import {LocationContext} from "./location";
+import {copyObject} from "../utils";
 
 const defaults = {
-  focusPath: null,
-  setFocusPath: () => {},
+  focus: {
+    path: [],
+    field: null,
+    on: null,
+    last: null,
+  },
+  changeFocus: () => {},
   availableFocusPaths: [],
 };
 
 const findUniqueFieldNames = (resourceObjects) => {
```

Comment:
This method gets called for every field on every render, wonder if it could be memoized instead.